src/FffGcodeWriter.cpp

```diff
@@ -2524,22 +2523,22 @@ bool FffGcodeWriter::processSupportInfill(const SliceDataStorage& storage, Layer
                                    support_line_width, support_line_distance_here, current_support_infill_overlap - (density_idx == max_density_idx ? 0 : wall_line_count * support_line_width),
                                    infill_multiplier, support_infill_angle, gcode_layer.z, support_shift,
                                    max_resolution, max_deviation,
-                                   (density_idx == max_density_idx ? wall_line_count : 0), infill_origin, support_connect_zigzags,
+                                   0 /*walls are generated somewhere else, so their layers aren't vertically combined*/, infill_origin, support_connect_zigzags,
                                    use_endpieces, skip_some_zags, zag_skip_count, pocket_size);
-                infill_comp.generate(wall_toolpaths, support_polygons, support_lines, infill_extruder.settings, storage.support.cross_fill_provider);
+                infill_comp.generate(dummy_wall_toolpaths, support_polygons, support_lines, infill_extruder.settings, storage.support.cross_fill_provider);
             }
 
             setExtruder_addPrime(storage, gcode_layer, extruder_nr); // only switch extruder if we're sure we're going to switch
             gcode_layer.setIsInside(false); // going to print stuff outside print object, i.e. support
 
-            if (!wall_toolpaths.empty())
+            if (! part.wall_toolpaths.empty())
```

Comment:
If you assign the part.wall_toolpaths to the wall_toolpaths as suggested above. Don't forget to rollback these ;-)

Reply:
If I do that, I won't forget ;-)
... I hope.

Reply:
I thought so I have high regards for your mental capabilities

Reply:
Turns out I didn't forget.